src/FileMigrationHelper.php

```diff
@@ -56,27 +58,37 @@ public function run($base = null)
         Environment::increaseTimeLimitTo();
         Environment::increaseMemoryLimitTo();
 
+        $logger = Injector::inst()->get(LoggerInterface::class);
+
         // Loop over all files
-        $count = 0;
+        $i = $migrated = 0;
         $originalState = null;
         if (class_exists(Versioned::class)) {
             $originalState = Versioned::get_reading_mode();
             Versioned::set_stage(Versioned::DRAFT);
         }
 
-        foreach ($this->getFileQuery() as $file) {
+        $query = $this->getFileQuery();
+        $total = $query->count();
+        foreach ($query as $file) {
             // Bypass the accessor and the filename from the column
             $filename = $file->getField('Filename');
 
             $success = $this->migrateFile($base, $file, $filename);
+
+            if ($i % 100 === 0) {
+                $logger->info("Iterated $i out of $total files. Migrated $migrated files.");
+            }
+
+            $i++;
             if ($success) {
-                $count++;
+                $migrated++;
             }
```

Comment:
could add a log worning about the file that wasn't a success?

Reply:
I'm a bit worried this could cause problems if it skips a lot of files. We don't want to overload the log output.

Reply:
though users would be able to control it with the log level?

Reply:
Alright, I'll add a warning.